### ISSUE TITLE
[a11y]: Progress Indicator does not support step labels that wrap AND have a secondaryLabel

### ISSUE THREAD

**Issue by embrown-ibm** [carbon-design-system/carbon#11000]
### Package

carbon-components, carbon-components-react

### Browser

Chrome, Safari, Firefox, Edge

### Operating System

MacOS

### Package version

v10.54.0

### React version

v7.54.2

### Automated testing tool and ruleset

Problem found by manual tester

### Description

I asked about the remaining problem (and this one) here...
https://ibm-cloudplatform.slack.com/archives/C046Y0YUD/p1647002467687969

I am using a `<ProgressIndicator>` .  My application supports 10 languages so we have a situation where the labelText for some of the `<ProgressStep>` components is long.  So I was able to easily fix this because the default for the step text (**bx--progress-label**) was `white-space: nowrap;` which I changed to `white-space: normal`
But the problem comes in when I have specified secondaryLabelText (**bx--progress-optional**) for a step and the step text wraps or is long and shows an overflow.  The problem is that the secondaryLabelText has an absolute position which assumes that the labelText will _always_ be one line of text or never wrapping.  With our application, some of the text of the step labels tend to be long and they overflow (which by default shows an ellipsis) due to translation.  If I modify the style of the `<ProgressStep>` labelText so it wraps, then the secondaryLabelText gets overlapped.

### Normal circumstances
<img width="1022" alt="image" src="https://user-images.githubusercontent.com/25669267/158640841-84b9e364-4679-4ae0-a818-7c205179cdc3.png">

### Text too long
_by default the step label text does not wrap, overflows with an ellipsis_
<img width="1032" alt="image" src="https://user-images.githubusercontent.com/25669267/158641180-03b361c9-5f80-4b34-bd32-26bd7a826ada.png">

### Change step label text to wrap
_if I set the `<ProgressStep>` labelText to `white-space: normal` the secondaryLabelText is overwritten_
![image](https://user-images.githubusercontent.com/25669267/158641546-e0b2c495-45cc-455e-adac-04a53dddbb5c.png)

### CodeSandbox example

https://codesandbox.io/s/compassionate-ellis-bg7fgr

### Steps to reproduce

- Create a `<ProgressIndicator>` which has several `<ProgressStep>` components as children with at least one of the steps having secondary label text.
- Either create the step label text to be really long or resize/reduce the size of your browser until the step label text extends into the area of the adjacent step label.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by embrown-ibm** [comment]
In my opinion, the **secondaryLabelText** of the `<ProgressStep>` should be positioned relative to the bottom of the **labelText**.  Currently it seems it is positioned relative to the overall `<ProgressIndicator>`

**Comment by dakahn** [comment]
@carbon-design-system/design have we given any thought to wrapping label names and how this should ideally be solved?

**Comment by abbeyhrt** [comment]
I think the intended solution is to use a `Tooltip` if `label` is too long to show all of it, through the `renderLabel` prop.  If you click on the second or third step in this example, you can see the tooltip being used. https://react.carbondesignsystem.com/?path=/story/components-progressindicator--default

The examples just show random text but in theory, you could have it equal the value of label so that users have a way to see the full text. 

Something like: 
```
<ProgressStep
      label="Second step with tooltip"
      description="Step 2: Getting started with Carbon Design System"
      renderLabel={() => (
        <Tooltip
          direction="bottom"
          showIcon={false}
          triggerClassName={`${prefix}--progress-label`}
          triggerText={'Second step with tooltip'}
          tooltipId="tooltipId-0">
          <p>Second step with tooltip</p>
        </Tooltip>
      )}
    />
```

**Comment by embrown-ibm** [comment]
@abbeyhrt To solve the original problem, I have to add `overflow: hidden` to the **bx--progress-text** to prevent the extra text from overlapping the adjacent step label.  The default behavior is to show a native flyover tooltip when hovering the text anyway.  
You solution to add a `<Tooltip>` or `<TooltipDefinition>` is a bit extra because due to supporting 10 languages, we would have to put it on all steps for consistency although it is only needed in a few instances.  I will have to check with the tester to see if the additional tooltip is needed.
Thanx!!!
